pkg/apierror/apierror.go

```diff
@@ -0,0 +1,125 @@
+package apierror
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/grpc-ecosystem/grpc-gateway/runtime"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	apierrorpb "github.com/openbank/gunk/gunk/v1/apierror"
+)
+
+var baseDocURL string
+
+// APIError embedds APIError definition from openbank/gunk.
+type APIError struct {
+	apierrorpb.APIError
+}
+
+// Error returns the error message.
+func (a APIError) Error() string {
+	return a.Message
+}
+
+// GRPCStatus returns the gRPC status according to the APIError.
+func (a APIError) GRPCStatus() *status.Status {
+	s := status.New(codes.Code(a.GetCode()), a.GetMessage())
+	s, _ = s.WithDetails(&a.APIError)
+	return s
+}
+
+// MarshalJSON is the custom json marshaling for an APIError.
+func (a APIError) MarshalJSON() ([]byte, error) {
+	err := map[string]interface{}{
+		"message": a.GetMessage(),
+	}
+
+	if a.GetDocURL() != "" {
```

Comment:
Is this check necessary?

I think we should put doc_url to map regardless of its value, so we will have consistent structure

Reply:
hm yes, users should at least get the `baseURL` of the error page